frontend/apps/vft-extended/src/hooks/api.ts

```diff
@@ -0,0 +1,245 @@
+ *
+ * This is a complete set of React hooks for working with
+ * the Foundable Token Extended (VFT) contract in SailsJS-based apps.
+ *
+ * Includes examples:
+ *  - Transaction hooks (mint, burn, transfer)
+ *  - State queries (name, symbol, decimals, totalSupply, balanceOf)
+ *  - Contract event subscriptions (minted, burned, transferred, approval)
+ *
+ *
+ * For full documentation, see the wiki:
+ *   https://wiki.gear.foundation/docs/sails-js/react-hooks
+ *
+ * For the VFT contract standard and API reference, see:
+ *   https://wiki.gear.foundation/docs/examples/Standards/vft
+ * ---------------------------------------------------------
+ */
+
+import { useProgram, useSendProgramTransaction, useProgramQuery, useProgramEvent } from '@gear-js/react-hooks';
+
+import { ENV } from '../consts';
+
+import { SailsProgram } from './lib';
+import { TokenMetaQueries, TokenEventCallbacks } from './types.ts';
+
+/**
+ * Returns a contract program instance.
+ *
+ * This hook sets up the connection to your contract using its ABI (library) and on-chain id.
+ * The returned instance is passed to all other hooks in this file.
+ *
+ * Note: You can use another id (address) or ABI to connect to different contracts.
+ */
+export function useProgramInstance() {
+  return useProgram({
+    library: SailsProgram, // ABI of the contract (must match what is deployed)
+    id: ENV.CONTRACT, // Address (id) of the contract on-chain
+  });
+}
+
+/**
+ * Returns async token actions and their loading states.
+ *
+ * This hook prepares all common token operations: mint, burn, transfer.
+ * Each action is async and can be awaited; you get the current loading (pending) state for each.
+ *
+ * Note: Each action uses serviceName and functionName matching the contract API.
+ * You can easily add more actions if needed (see sendTransactionAsync usage).
+ */
+export function useTokenActions() {
+  const { data: program } = useProgramInstance();
+
+  // Hook for minting tokens
+  const { sendTransactionAsync: sendMintTransactionAsync, isPending: mintPending } = useSendProgramTransaction({
+    program,
+    serviceName: 'vft',
+    functionName: 'mint',
+  });
+
+  // Hook for burning tokens
+  const { sendTransactionAsync: sendBurnTransactionAsync, isPending: burnPending } = useSendProgramTransaction({
+    program,
+    serviceName: 'vft',
+    functionName: 'burn',
+  });
+
+  // Hook for transferring tokens
+  const { sendTransactionAsync: sendTransferTransactionAsync, isPending: transferPending } = useSendProgramTransaction({
+    program,
+    serviceName: 'vft',
+    functionName: 'transfer',
+  });
+
+  /**
+   * Async wrapper for minting tokens.
+   * @param to Recipient address (must be a hex string with 0x prefix)
+   * @param value Amount to mint (string, because JS can't safely handle big numbers)
+   */
+  const mint = async (to: `0x${string}`, value: string) => {
+    return sendMintTransactionAsync({
+      args: [to, value],
+    });
+  };
```

Comment:
If your purpose is to highlight the use cases of our API, would be better to do it via interfaces described in docs, and it's better to avoid adding additional abstraction layers.

More straightforward solution would look like this:

```js
function useSendMintTransaction() {
  const { data: program } = useProgramInstance();
  
  return useSendProgramTransaction({
    program,
    serviceName: 'vft',
    functionName: 'mint',
  });
} 
```